devU-api/src/entities/grader/grader.service.ts

```diff
@@ -0,0 +1,96 @@
+import submissionService from '../submission/submission.service'
+import submissionScoreService from '../submissionScore/submissionScore.service'
+import submissionProblemScoreService from '../submissionProblemScore/submissionProblemScore.service'
+import nonContainerAutograderService from '../nonContainerAutoGrader/nonContainerAutoGrader.service'
+import containerAutograderService from '../containerAutoGrader/containerAutoGrader.service'
+import assignmentProblemService from '../assignmentProblem/assignmentProblem.service'
+
+import { SubmissionScore, SubmissionProblemScore, ContainerAutoGrader } from 'devu-shared-modules'
+import { checkAnswer } from '../nonContainerAutoGrader/nonContainerAutoGrader.grader'
+import { serialize as serializeNonContainer } from '../nonContainerAutoGrader/nonContainerAutoGrader.serializer'
+import { serialize as serializeContainer } from '../containerAutoGrader/containerAutoGrader.serializer'
+
+export async function grade(id: number) {
+    const submission = await submissionService.retrieve(id)
+    if (!submission) return null
+
+    const assignmentId = submission.assignmentId
+
+    const content = JSON.parse(submission.content)
+    const form = content.form
+    const filepaths = content.filepaths //Using the field name that was written on the whiteboard for now
+
+    const nonContainerAutograders = await nonContainerAutograderService.listByAssignmentId(assignmentId)
+    const containerAutograders = await containerAutograderService.listByAssignmentId(assignmentId)
+    const assignmentProblems = await assignmentProblemService.list(assignmentId)
+
+    
+    var score = 0
+    var allScores = [] //This is the return value, the serializer parses it into a GraderInfo object for the controller to return
+
+    //Run Non-Container Autograders
+    for (const question in form) { 
+        const nonContainerGrader = nonContainerAutograders.find(grader => grader.question === question)
+        const assignmentProblem = assignmentProblems.find(problem => problem.problemName === question)
+        
+        if (nonContainerGrader && assignmentProblem) {
+            const problemScore = await checkAnswer(form[question], serializeNonContainer(nonContainerGrader)) //Should also return feedback in the future
+            score += problemScore
+
+            const problemScoreObj: SubmissionProblemScore = {
+                submissionId: id,
+                assignmentProblemId: assignmentProblem.id,
+                score: problemScore,
+                feedback: '' //grader doesn't currently return feedback
+            }
+            allScores.push(await submissionProblemScoreService.create(problemScoreObj))
+        }
+    }
```

Comment:
This is fine for now, but it does assume that there's <=1 grader per question. In most cases this _should_ be true, but we shouldn't rely on it. When we expand to Container graders we can update